src/main/kotlin/build/buf/intellij/resolve/RefreshAdditionalBufRoots.kt

```diff
@@ -43,33 +48,65 @@ class RefreshAdditionalBufRoots : StartupActivity {
                         }
                     }
                     if (bufLockChanged) {
-                        updateUserRoots(project)
+                        updateUserRoots(project, false)
                     }
                 }
             },
         )
-        updateUserRoots(project)
+        updateUserRoots(project, true)
     }
 
-    private fun updateUserRoots(project: Project) {
-        DumbService.getInstance(project).runWhenSmart {
-            val existingModuleKeys = project.getUserData(BufModuleKeysUserData.KEY)?.moduleKeys ?: emptySet()
-            val newModuleKeys = hashSetOf<ModuleKey>()
-            // TODO: This still can return stale data after VFS changes.
-            // Unclear how to ensure we run this after indexing has completed.
-            // There don't appear to be any message bus changes that fire when indexing completes.
-            for (moduleKey in BufIndexes.getProjectModuleKeys(project)) {
-                newModuleKeys.add(moduleKey)
+    private fun updateUserRoots(project: Project, initial: Boolean) {
+        ApplicationManager.getApplication().executeOnPooledThread {
+            DumbService.getInstance(project).runWhenSmart {
+                if (initial) {
+                    // Trigger initial 'buf build' on all modules to trigger download of dependencies
+                    buildAllModules(project)
```

Comment:
Testing with the latest version of the buf CLI and an empty cache (after `buf registry cc`), this fixes the issue where dependency roots aren't shown until an edit of a .proto file (triggering a format/lint/breaking call) and the project is reloaded.